### PR TITLE
Grab AWS group policies on top of user

### PR DESCRIPTION
**Pull request by stxue1**
Resolves possible AWS group permissions issue from #4485 

## Changelog Entry
To be copied to the [draft changelog](https://github.com/DataBiosphere/toil/wiki/Draft-Changelog) by merger:

 * Grab AWS group policies associated with a user and add a unit test for collecting AWS permissions

## Reviewer Checklist

 * [ ] Make sure it is coming from `issues/XXXX-fix-the-thing` in the Toil repo, or from an external repo.
    * [ ] If it is coming from an external repo, make sure to pull it in for CI with:
        ```
        contrib/admin/test-pr otheruser theirbranchname issues/XXXX-fix-the-thing
        ```
    * [ ] If there is no associated issue, [create one](https://github.com/DataBiosphere/toil/issues/new).
* [ ] Read through the code changes. Make sure that it doesn't have:
    * [ ] Addition of trailing whitespace.
    * [ ] New variable or member names in `camelCase` that want to be in `snake_case`.
    * [ ] New functions without [type hints](https://docs.python.org/3/library/typing.html).
    * [ ] New functions or classes without informative docstrings.
    * [ ] Changes to semantics not reflected in the relevant docstrings.
    * [ ] New or changed command line options for Toil workflows that are not reflected in `docs/running/{cliOptions,cwl,wdl}.rst` 
    * [ ] New features without tests.
* [ ] Comment on the lines of code where problems exist with a review comment. You can shift-click the line numbers in the diff to select multiple lines.
* [ ] Finish the review with an overall description of your opinion.

## Merger Checklist

* [ ] Make sure the PR passes tests.
* [ ] Make sure the PR has been reviewed **since its last modification**. If not, review it.
* [ ] Merge with the Github "Squash and merge" feature.
    * [ ] If there are multiple authors' commits, add [Co-authored-by](https://github.blog/2018-01-29-commit-together-with-co-authors/) to give credit to all contributing authors.
* [ ] Copy its recommended changelog entry to the [Draft Changelog](https://github.com/DataBiosphere/toil/wiki/Draft-Changelog).
* [ ] Append the issue number in parentheses to the changelog entry.

### CODE REVIEW

File: src/toil/lib/aws/iam.py [hunk 1]
```diff
@@ -232,6 +232,16 @@ def get_policy_permissions(region: str) -> AllowedActionCollection:
...
         allowed_actions = add_to_action_collection(allowed_actions, user_inline_policies)
 
+        # grab group policies associated with the user
+        groups = iam.list_groups_for_user(UserName=user['User']['UserName'])
+        for group in groups["Groups"]:
+            list_policies = iam.list_group_policies(GroupName=group['GroupName'])
+            attached_policies = iam.list_attached_group_policies(GroupName=group['GroupName'])
+            group_attached_policies = allowed_actions_attached(iam, attached_policies['AttachedPolicies'])
+            allowed_actions = add_to_action_collection(allowed_actions, group_attached_policies)
+            group_inline_policies = allowed_actions_users(iam, list_policies['PolicyNames'], user['User']['UserName'])
```
Comment (adamnovak) [inline src/toil/lib/aws/iam.py:242]: This is going to need to be a new `allowed_actions_groups` probably that gets the inline *group* policies from the *group*, not the inline user policies from the user.

**Review by adamnovak — CHANGES_REQUESTED** [review]
I think one part of the user logic isn't generalized properly to groups. Also I think we can add tests for this with Moto.

**Review by adamnovak — APPROVED** [review]
I think this is pretty good! I am just going to add the missing docstring and then I think we can merge it.

**Review by DailyDreaming — APPROVED** [review]
LGTM.  Only minor comments.

**Pull request merged**